data/data/azure/vnet/main.tf

```diff
@@ -43,15 +43,52 @@ data "azurerm_resource_group" "network" {
   name = var.azure_network_resource_group_name
 }
 
+data "azurerm_key_vault" "keyvault" {
+  count = var.azure_keyvault_name != "" ? 1 : 0
+
+  name                = var.azure_keyvault_name
+  resource_group_name = var.azure_keyvault_resource_group
+}
+
+data "azurerm_key_vault_key" "keyvault_key" {
+  count = var.azure_keyvault_name != "" ? 1 : 0
+
+  name         = var.azure_keyvault_key_name
+  key_vault_id = data.azurerm_key_vault.keyvault[0].id
+}
+
+data "azurerm_user_assigned_identity" "keyvault_identity" {
+  count = var.azure_keyvault_name != "" ? 1 : 0
+
+  resource_group_name = var.azure_keyvault_resource_group
+  name                = var.azure_user_assigned_identity_key
+}
+
 resource "azurerm_storage_account" "cluster" {
   name                            = "cluster${var.random_storage_account_suffix}"
   resource_group_name             = data.azurerm_resource_group.main.name
   location                        = var.azure_region
-  account_tier                    = "Standard"
+  account_tier                    = var.azure_keyvault_name != "" ? "Premium" : "Standard"
   account_replication_type        = "LRS"
   min_tls_version                 = contains(local.environments_with_min_tls_version, var.azure_environment) ? "TLS1_2" : null
-  allow_nested_items_to_be_public = false
+  allow_nested_items_to_be_public = var.azure_keyvault_name != "" ? true : false
```

Comment:
Do you know why this is needed?

Reply:
Probably could remove it. @jhixson74 should we remove this?

Reply:
This is required to allow blob access

Reply:
Oh, so it has to be public because it's in a different subscription. That's fine however, since to access it you need the the encryption key.